tests/Fixer/ClassNotation/FinalPublicMethodForAbstractClassFixerTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ClassNotation;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ClassNotation\FinalPublicMethodForAbstractClassFixer
+ */
+final class FinalPublicMethodForAbstractClassFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected PHP source code
+     * @param null|string $input    PHP source code
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        $original = $fixed = $this->getClassElementStubs();
+        $fixed = str_replace('public function f1', 'final public function f1', $fixed);
+        $fixed = str_replace('public static function f4', 'final public static function f4', $fixed);
+        $fixed = str_replace('static public function f7', 'final static public function f7', $fixed);
+
+        return [
+            'regular-class' => ["<?php class MyClass { ${original} }"],
+            'final-class' => ["<?php final class MyClass { ${original} }"],
+            'trait' => ["<?php trait MyClass { ${original} }"],
+            'interface' => [
+                '<?php interface MyClass {
+                    public function f1();
+                    public static function f4();
+                    static public function f7();
+                }',
+            ],
+            'magic-methods' => [
```

Comment:
please add a test with a magic method in different casing, like `__CLONE(){}`

Reply:
Good catch, done :+1: